api/services/organization/Organization.js

```diff
@@ -0,0 +1,256 @@
+const {
+  Organization,
+  OrganizationRole,
+  Role,
+  UAAIdentity,
+  User,
+} = require('../../models');
+const { CustomError } = require('../../utils/validators');
+const UAAClient = require('../../utils/uaaClient');
+
+/**
+ * Not typing the Sequelize stuff...
+ *
+ * @typedef {object} UserType
+ * @typedef {object} OrgType
+ * @typedef {object} UAAIdentityType
+ */
+
+/**
+ * @typedef {object} UAAInvite
+ * @property {string} email
+ * @property {string} userId
+ * @property {string} origin
+ * @property {boolean} success
+ * @property {string=} errorCode
+ * @property {string=} errorMessage
+ * @property {string=} inviteLink
+ */
+
+function throwError(message) {
+  throw new CustomError(message, 422);
+}
+
+module.exports = {
+  /**
+   * @param {UAAIdentityType} currentUserUAAIdentity
+   *
+   * Refreshes and saves user access token
+   */
+  async refreshToken(currentUserUAAIdentity) {
+    const uaaClient = new UAAClient();
+
+    const { accessToken, refreshToken } = await uaaClient.refreshToken(
+      currentUserUAAIdentity.refreshToken
+    );
+
+    await currentUserUAAIdentity.update({ accessToken, refreshToken });
+
+    return accessToken;
+  },
+
+  /**
+   * @param {UAAIdentityType} currentUserUAAIdentity
+   * @param {string} targetUserEmail - the email address of the user to invite
+   * @returns {Promise<UAAInvite>}
+   */
+  async inviteUAAUser(currentUserUAAIdentity, targetUserEmail) {
+    const accessToken = await this.refreshToken(currentUserUAAIdentity);
+
+    const uaaClient = new UAAClient();
+    return uaaClient.inviteUserToUserGroup(targetUserEmail, accessToken);
+  },
+
+  /**
+   * @param {UAAIdentityType} currentUserUAAIdentity
+   */
+  async isUAAAdmin(currentUserUAAIdentity) {
+    const uaaClient = new UAAClient();
+    return uaaClient.verifyUserGroup(currentUserUAAIdentity.uaaId, ['pages.admin']);
+  },
+
+  /**
+   * @param {string} uaaEmail
+   * @returns {Promise<UserType | null>}
+   */
+  findUserByUAAIdentity(uaaEmail) {
+    return User.findOne({
+      include: [{
+        model: UAAIdentity,
+        where: { email: uaaEmail },
+        required: true,
+      }],
+    });
+  },
+
+  /**
+   * @param {UAAIdentityType} currentUserUAAIdentity
+   * @param {string} targetUserEmail - the email address of the user to invite
+   * @param {string=} targetUserGithubUsername - the github username of the user to invite,
+   * if they are a current Federalist user
+   *
+   * @returns {Promise<[UserType, (UAAInvite | undefined)]>}
+   */
+  async findOrCreateUAAUser(currentUserUAAIdentity, targetUserEmail, targetUserGithubUsername = '') {
+    let user = await this.findUserByUAAIdentity(targetUserEmail);
+
+    if (user) {
+      return [user];
+    }
+
+    const invite = await this.inviteUAAUser(currentUserUAAIdentity, targetUserEmail);
+
+    user = await User.findOne({
+      where: { username: targetUserGithubUsername && targetUserGithubUsername.toLowerCase() },
```

Comment:
all github usernames are inserted as lowercase ... curious, did you experience otherwise?

Reply:
doh! ... you are 💯  correct!